sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -286,134 +216,80 @@ def schedule(self, schedule_time):
             self.properties.message_id = str(uuid.uuid4())
         if not self.message.annotations:
             self.message.annotations = {}
-        self.message.annotations[types.AMQPSymbol(self._x_OPT_SCHEDULED_ENQUEUE_TIME)] = schedule_time
+        self.message.annotations[types.AMQPSymbol(_X_OPT_SCHEDULED_ENQUEUE_TIME)] = schedule_time
 
-    def renew_lock(self):
-        """Renew the message lock.
 
-        This will maintain the lock on the message to ensure
-        it is not returned to the queue to be reprocessed. In order to complete (or otherwise settle)
-        the message, the lock must be maintained. Messages received via ReceiveAndDelete mode are not
-        locked, and therefore cannot be renewed. This operation can also be performed as a threaded
-        background task by registering the message with an `azure.servicebus.AutoLockRenew` instance.
-        This operation is only available for non-sessionful messages.
+class BatchMessage(object):
+    """A batch of messages.
 
-        :raises: TypeError if the message is sessionful.
-        :raises: ~azure.servicebus.common.errors.MessageLockExpired is message lock has already expired.
-        :raises: ~azure.servicebus.common.errors.MessageAlreadySettled is message has already been settled.
-        """
-        if hasattr(self._receiver, 'locked_until'):
-            raise TypeError("Session messages cannot be renewed. Please renew the Session lock instead.")
-        self._is_live('renew')
-        token = self.lock_token
-        if not token:
-            raise ValueError("Unable to renew lock - no lock token found.")
+    Sending messages in a batch is more performant than sending individual message.
+    BatchMessage helps you create the maximum allowed size batch of `Message` to improve sending performance.
 
-        expiry = self._receiver._renew_locks(token)  # pylint: disable=protected-access
-        self._expiry = datetime.datetime.fromtimestamp(expiry[b'expirations'][0]/1000.0)
+    Use the `add` method to add messages until the maximum batch size limit in bytes has been reached -
+    at which point a `ValueError` will be raised.
 
-    def complete(self):
-        """Complete the message.
+    **Please use the create_batch method of ServiceBusSender
```

Comment:
Candidly, in all my writing of unit tests, I never once saw this comment.  While I'm very willing to attribute that to my own blindness, I was using BatchMessage() manually without issue, and I fear other users will do the same.

I assume this is largely because of the max_size_bytes logic, but if that's a standard value 99% of the time, I wonder if there's any way we can do this that's robust with respect to the naïve approach.

Reply:
(Closing this for now as we've brought the discussion up with the other SB folks and will resolve this via a combination of architectural discussion and user studies)